pulp_python/app/utils.py

```diff
@@ -187,6 +192,25 @@ def get_project_metadata_from_file(filename):
     return metadata
 
 
+def compute_metadata_sha256(filename: str) -> str:
+    """
+    Compute SHA256 hash of the metadata file from a Python package.
+
+    Returns SHA256 hash or empty string if metadata cannot be extracted.
+    """
+    if not filename.endswith(".whl"):
+        return ""
+    try:
+        with zipfile.ZipFile(filename, "r") as f:
+            for file_path in f.namelist():
+                if file_path.endswith(".dist-info/METADATA"):
+                    metadata_content = f.read(file_path)
+                    return hashlib.sha256(metadata_content).hexdigest()
+    except (zipfile.BadZipFile, KeyError, OSError):
+        pass
+    return ""
```

Comment:
I wouldn't think we would ever want to return empty string?  Maybe "real checksum, else `None`"?  The field was made nullable, so empty string would be like a 2nd null value type.

Reply:
Good catch! I also removed `package["metadata_sha256"] = package.get("metadata_sha256") or ""`, because the `package` already contains `metadata_sha256` - it comes from `package = parse_project_metadata(project)`